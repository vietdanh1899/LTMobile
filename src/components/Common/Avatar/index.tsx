import React from 'react';
import { StyleSheet } from 'react-native';
import {
  Avatar as EAvatar,
  AvatarProps as EAvatarProps,
} from 'react-native-elements';

export interface AvatarProps extends EAvatarProps {
  margin?: number;
  marginTop?: number;
  marginBottom?: number;
  marginLeft?: number;
  marginRight?: number;
  marginHorizontal?: number;
  marginVertical?: number;
  backgroundColor?: string;
}

class Avatar extends React.PureComponent<AvatarProps> {
  render() {
    const {
      margin,
      marginTop,
      marginBottom,
      marginLeft,
      marginRight,
      marginHorizontal,
      marginVertical,
      backgroundColor,
      overlayContainerStyle: overlayContainerStyleProp,
      containerStyle: containerStyleProp,
    } = this.props;

    const overlayContainerStyle = StyleSheet.flatten([
      { backgroundColor },
      overlayContainerStyleProp,
    ]);

    const containerStyle = StyleSheet.flatten([
      {
        margin,
        marginTop,
        marginBottom,
        marginLeft,
        marginRight,
        marginHorizontal,
        marginVertical,
      },
      containerStyleProp,
    ]);

    return (
      <EAvatar
        {...this.props}
        overlayContainerStyle={overlayContainerStyle}
        containerStyle={containerStyle}
      />
    );
  }
}

export default Avatar;
