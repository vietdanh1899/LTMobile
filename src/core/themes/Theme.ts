import {
  bottomNavigationBarHeight,
  headerPaddingHorizontal,
  bodyPaddingHorizontal,
  shadowViewLight,
  shadowViewDark,
  darkComponentColor,
  lightComponentColor,
} from './ThemeComponent/Common/CommonProps';
import {
  Color,
  darkPlatform,
  lightPlatform,
  darkPrimaryColor,
  darkSecondaryColor,
  lightPrimaryColor,
  lightSecondaryColor,
} from './ThemeComponent/Common/Color';
import { darkText, lightText } from './ThemeComponent/Text';
import { darkButton, lightButton } from './ThemeComponent/Button';
import { darkHeader, lightHeader } from './ThemeComponent/Header';
import { darkInput, lightInput } from './ThemeComponent/Input';
import { darkCheckBox, lightCheckBox } from './ThemeComponent/ListCheckBox';
import { darkDropdown, lightDropdown } from './ThemeComponent/Dropdown';
import { darkModal, lightModal } from './ThemeComponent/Modal';
import { lightParallaxScrollView, darkParallaxScrollView } from './ThemeComponent/ParallaxScrollView';

const ThemeEnum = {
  DARK: 'dark',
  LIGHT: 'light',
};

/**
 * Theme
 */
export const darkTheme = {
  key: ThemeEnum.DARK,
  colors: {
    primary: darkPrimaryColor,
    secondary: darkSecondaryColor,

    white: Color.white,
    grey1: Color.grey1,
    grey2: Color.grey2,
    grey3: Color.grey3,
    grey4: Color.grey4,
    grey5: Color.grey5,
    grey6: Color.grey6,
    grey7: Color.grey7,
    grey8: Color.grey8,
    grey10: Color.grey10,
    black: Color.black,

    greyOutline: '#bbb',
    searchBg: '#303337',

    success: Color.green,
    error: Color.red,
    warning: Color.yellow,
    disabled: 'hsl(208, 8%, 90%)',
    // Darker color if hairlineWidth is not thin enough
    divider: Color.grey6,
    border: Color.grey9,
    platform: darkPlatform,

    /**
    * Component Color
    */
    bgColor: darkComponentColor.bgColor,
    bgColorSecondary: darkComponentColor.bgColorSecondary,
    textColor: darkComponentColor.textColor,
    textColorSecondary: darkComponentColor.textColorSecondary,
  },
  /**
   * Component
   */
  bottomNavigationBarHeight,
  headerPaddingHorizontal,
  bodyPaddingHorizontal,
  shadowView: shadowViewDark,

  Text: darkText,
  Button: darkButton,
  Header: darkHeader,
  Input: darkInput,
  CheckBox: darkCheckBox,
  Dropdown: darkDropdown,
  AppModal: darkModal,
  ParallaxScrollView: darkParallaxScrollView,
};

export const lightTheme = {
  key: ThemeEnum.LIGHT,
  colors: {
    primary: lightPrimaryColor,
    secondary: lightSecondaryColor,

    white: Color.white,
    grey1: Color.grey1,
    grey2: Color.grey2,
    grey3: Color.grey3,
    grey4: Color.grey4,
    grey5: Color.grey5,
    grey6: Color.grey6,
    grey7: Color.grey7,
    grey8: Color.grey8,
    grey10: Color.grey10,
    black: Color.black,

    greyOutline: '#bbb',
    searchBg: '#303337',

    listItemBg: Color.white,
    success: Color.green,
    error: Color.red,
    warning: Color.yellow,
    disabled: 'hsl(208, 8%, 90%)',
    // Darker color if hairlineWidth is not thin enough
    divider: Color.grey2,
    border: Color.grey2,
    platform: lightPlatform,

    /**
     * Component Color
     */
    bgColor: lightComponentColor.bgColor,
    bgColorSecondary: lightComponentColor.bgColorSecondary,
    textColor: lightComponentColor.textColor,
    textColorSecondary: lightComponentColor.textColorSecondary,
  },

  /**
   * Component
   */
  bottomNavigationBarHeight,
  headerPaddingHorizontal,
  bodyPaddingHorizontal,
  shadowView: shadowViewLight,

  Text: lightText,
  Button: lightButton,
  Header: lightHeader,
  Input: lightInput,
  CheckBox: lightCheckBox,
  Dropdown: lightDropdown,
  AppModal: lightModal,
  ParallaxScrollView: lightParallaxScrollView,
};

export { Color };
export default lightTheme;
