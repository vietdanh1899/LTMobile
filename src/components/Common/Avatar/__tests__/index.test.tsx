import React from 'react';
import {
  basicTest, testStyleProps, EnumComponent,
} from '@utils/testHelper';
import { Color } from '@themes/Theme';
import Avatar from '..';

const target = <Avatar />;
const type = EnumComponent.SINGLE;

describe('Avatar Component', () => {
  basicTest(target, type);

  it('should apply all style-related props', () => {
    const props = {
      margin: 10,
      marginVertical: 10,
      marginHorizontal: 10,
      marginTop: 10,
      marginBottom: 10,
      marginLeft: 10,
      marginRight: 10,
    };
    testStyleProps(target, props, type, 'containerStyle');
    const props2 = {
      backgroundColor: Color.orange,
    };
    testStyleProps(target, props2, type, 'overlayContainerStyle');
  });
});
