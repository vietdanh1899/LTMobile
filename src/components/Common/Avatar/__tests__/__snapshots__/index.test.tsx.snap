// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar Component should render without issues 1`] = `
<Themed.Avatar
  containerStyle={
    Object {
      "margin": undefined,
      "marginBottom": undefined,
      "marginHorizontal": undefined,
      "marginLeft": undefined,
      "marginRight": undefined,
      "marginTop": undefined,
      "marginVertical": undefined,
    }
  }
  overlayContainerStyle={
    Object {
      "backgroundColor": undefined,
    }
  }
/>
`;
